.github/workflows/prepare-release-branch.yml

```diff
@@ -0,0 +1,92 @@
+
+  create-pull-request-against-release-branch:
+    needs: prepare-release-branch
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{ needs.prepare-release-branch.outputs.release-branch-name }}
+
+      - name: Remove SNAPSHOT from version
+        run: |
+          v=$(grep -Eo "[0-9]+.[0-9]+.[0-9]+-SNAPSHOT" version.gradle.kts | sed 's/-SNAPSHOT//')
+          .github/scripts/update-versions.sh "$v-SNAPSHOT" "$v-alpha-SNAPSHOT" "$v" "$v-alpha"
+
+      - name: Bump download link version
+        run: |
+          new_version=$(grep -Eo "[0-9]+.[0-9]+.[0-9]+" version.gradle.kts | head -1)
+          sed -Ei "s,https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v[0-9]+.[0-9]+.[0-9]+/,https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v$new_version/," README.md
+
+      - name: Setup git name
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+
+      - name: Create pull request
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          msg="Prepare release branch ${{ needs.prepare-release-branch.outputs.release-branch-name }}"
+          git add -u
+          git commit -m "$msg"
+          git push origin HEAD:prepare-release-branch-${{ needs.prepare-release-branch.outputs.release-branch-name }}
+          gh pr create --title "$msg" \
+                       --body "$msg" \
+                       --head prepare-release-branch-${{ needs.prepare-release-branch.outputs.release-branch-name }} \
+                       --base ${{ needs.prepare-release-branch.outputs.release-branch-name }}
+
+  create-pull-request-against-main:
+    needs: prepare-release-branch
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+
+      - name: Bump SNAPSHOT version
+        run: |
+          v=$(grep -Eo "[0-9]+.[0-9]+.[0-9]+-SNAPSHOT" version.gradle.kts | sed 's/-SNAPSHOT//')
+          if [[ $v =~ ([0-9]+).([0-9]+).0 ]]; then
+            major="${BASH_REMATCH[1]}"
+            minor="${BASH_REMATCH[2]}"
+          else
+            echo "unexpected version: $v"
+            exit 1
+          fi
+          bump="$major.$((minor + 1)).0"
+          .github/scripts/update-versions.sh "$v-SNAPSHOT" "$v-alpha-SNAPSHOT" "$bump-SNAPSHOT" "$bump-alpha-SNAPSHOT"
+
+      - name: Setup git name
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+
+      - name: Create pull request
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          msg="Bump snapshot version"
+          git add -u
+          git commit -m "$msg"
+          git push origin HEAD:bump-snapshot-version
+          gh pr create --title "$msg" --body "$msg" --head bump-snapshot-version --base main
```

Comment:
I just noticed the `**/**` CNCF branch protection rule seems to be gone, did we accidentally delete it?

Though either way, even if we can make it work for release branch, this PR creation to `main` looks like it'll be problematic due to CLA check :(

Reply:
Though we do explicitly have the checkbox to let us merge even if status checks aren't all passing. So that could be our workaround

Reply:
we have branch protection on both `main` and `v**`, so CLA check will be a problem for both PRs 😞

Reply:
it looks there are [complexities to approving github-actions bot account for EasyCLA](https://github.com/open-telemetry/community/issues/809#issuecomment-905869941) and so that path could take some time

it seems that the current recommendation is to use a Personal Access Token generated by one of the maintainers: https://github.com/open-telemetry/community/blob/main/docs/using-github-extensions.md#writing-your-github-actions-pipelines

Reply:
reverted back to using github-actions bot for now at least